neothesia/src/scene/menu_scene/iced_menu/tracks.rs

```diff
@@ -78,17 +85,35 @@ impl Page for TracksPage {
                     let color = &ctx.config.color_schema[color_id].base;
                     iced_core::Color::from_rgb8(color.0, color.1, color.2)
                 };
+                let instrument_id = track
+                    .programs
+                    .last()
+                    .map(|p| p.program as usize)
+                    .unwrap_or(0);
 
                 let name = if track.has_drums && !track.has_other_than_drums {
                     "Percussion"
                 } else {
-                    let instrument_id = track
-                        .programs
-                        .last()
-                        .map(|p| p.program as usize)
-                        .unwrap_or(0);
                     midi_file::INSTRUMENT_NAMES[instrument_id]
                 };
+                
+   
+
+                // Check if the instrument is Grand Piano (id 0, 1, or 2)
+                let hand_info = if instrument_id <= 2 {
```

Comment:
We need to add `track.has_drums && !track.has_other_than_drums` here (it would be the best to move the one from line 94 to a variable to not duplicate this logic). The id check is not enough (probably because midi track 9 has special drum meaning)

eg. in one file I got
![image](https://github.com/PolyMeilex/Neothesia/assets/20758186/dd906577-5790-4e28-b82b-791e6d561dcc)

Reply:
thats weird. i added  
let hand_info = if instrument_id <= 2 && name != "Percussion" {